### PR TITLE
[TS-255] Display Positions in other dashboard 

### PR DESCRIPTION
**Pull request by tungleu**
1, Changed /other-dashboard into /dashboard/:userId
Use this for example `http://localhost:8081/dashboard/61d9304747dcc156a8042003`
2, The table in the dashboards now displays real data populated from Alpaca (User must be connected to Alpaca in order to see) 
3, `Position` and `Profile` components are now dynamic, I personally prefer to pass the list of positions as props than passing the userId. I couldn't display it with the way the table is coded for some reason, any help would be great 👼 
image

### CODE REVIEW

File: src/pages/OtherDashboard/OtherDashboard.vue [hunk 1]
```diff
@@ -96,17 +97,69 @@ export default {
           company: "Voyager",
           purchased: false,
           when: "Today",
-          today: true,
-        },
+          today: true
+        }
+      ],
+      stocks: [
```
Comment (GadoliniumPotassium) [inline src/pages/OtherDashboard/OtherDashboard.vue:103]: Why are you keeping this, I'd suggest you remove the commented out code (unless you want to keep as a structure reminder.

File: src/pages/OtherDashboard/OtherDashboard.vue [hunk 2]
```diff
@@ -96,17 +97,69 @@ export default {
...
+        // },
+        // {
+        //   symbol: "FSRLMAOOO",
+        //   positionSize: 240,
+        //   date: new Date("10/13/2021").toLocaleString(),
+        //   profitLoss: 67.3,
+        //   verified: true
+        // }
       ],
+      userInfo: {
```
Comment (GadoliniumPotassium) [inline src/pages/OtherDashboard/OtherDashboard.vue:126]: Why are you keeping the raw user data if its dynamic now

File: src/services/User.service.js [hunk 3]
```diff
@@ -1,20 +1,55 @@
...
+
+    return userPortfolioData;
+  }
+
+  async getUserInfo(userID) {
+    let userProfileData = null;
+    await axios
+      .get("/userInfo/" + userID)
+      .then(function(res) {
+        // console.log("User info success");
```
Comment (GadoliniumPotassium) [inline src/services/User.service.js:43]: You can remove the commented out code (Code smell)

File: src/pages/OtherDashboard/OtherDashboard.vue [hunk 4]
```diff
@@ -96,17 +97,35 @@ export default {
...
+      positions: []
     };
   },
+  created() {
+    this.initialize();
+  },
+  methods: {
+    async initialize() {
+      this.info = {
+        ...(await UserService.getUserInfo(this.userId)),
```
Comment (kennguyen0303) [inline src/pages/OtherDashboard/OtherDashboard.vue:113]: Have you tried the `undefined` value here? I'm not sure if `undefined` could causes error for the `...` operator or not

**Review by kennguyen0303 — CHANGES_REQUESTED** [review]
Some Javascript minor tweaks. The rest looks good

File: src/services/User.service.js [hunk 5]
```diff
@@ -1,20 +1,53 @@
...
+      return { success: data.success };
+    } catch (err) {
+      console.log(err);
+      return { success: false, message: err.response.data.message };
+    }
+  }
+
+  async getPositions(userID) {
+    let userPortfolioData = null;
+    await axios
```
[the remark this answers is not in the document]
Comment (tungleu) [reply src/services/User.service.js:21]: its not fetching when i removed it :flushed:

File: src/pages/OtherDashboard/OtherDashboard.vue (hunk 4 above)
Comment (tungleu) [reply src/pages/OtherDashboard/OtherDashboard.vue:113]: yep it wont cause any errors, I added a render condition if the userId doesn't exist

File: cypress/fixtures/url.json [hunk 6]
```diff
@@ -3,7 +3,7 @@
 	"HOSTED_BACKEND_URL": "https://tradeshare-backend-for-test.herokuapp.com/api/check",
 	"SIGN_UP": "/signup",
 	"LOGIN": "/login",
-	"OTHER_DASHBOARD": "/other-dashboard",
+	"OTHER_DASHBOARD": "/dashboard/61d9304747dcc156a8042003",
```
[the remark this answers is not in the document]
Comment (tungleu) [reply cypress/fixtures/url.json:6]: https://stackoverflow.com/questions/36962601/node-js-how-to-use-environment-variables-in-json-file

i'm aware of that but i don't think we can use environment variables in a JSON file?

**Review by rileykc98 — CHANGES_REQUESTED** [review]
Looks like profile tests are still failing. Perhaps the test user needs to be configured so that the now dynamic components display properly.

**Review by itsShaVawn — CHANGES_REQUESTED** [review]
URL for the other dashboard should be `/username/dashboard` instead of `/dashboard/randomsequenceoflettersandnumbers`

**Comment by tungleu** [comment]
> URL for the other dashboard should be `/username/dashboard` instead of `/dashboard/randomsequenceoflettersandnumbers`

will address it next sprint since it's not possible right now

**Comment by GadoliniumPotassium** [comment]
I think the next step is to deal with the conflicting files before we move on here

**Review by kennguyen0303 — APPROVED** [review]
lgtm

**Review by GadoliniumPotassium — APPROVED** [review]

**Pull request merged**